orcid-web/src/main/java/org/orcid/frontend/web/controllers/BaseController.java

```diff
@@ -353,7 +358,10 @@ protected void logoutCurrentUser(HttpServletRequest request, HttpServletResponse
 		}
 		if (authentication != null && authentication.isAuthenticated()) {
 			new SecurityContextLogoutHandler().logout(request, response, authentication);
```

Comment:
It gets set to null when logout is called. Then it gets generated by CsrfFilter.java on the next request. I think that is why it works when there is redirect.
In the case where there is no redirect, it is set to null, but the old token (associated with the previous session) is still set as a request attribute. It is then rendered in the page header, and used in subsequent ajax calls, but is invalid.